File: GalileoX_CAAD002X_Android_App_Development_for_Beginners/DatabaseExample_BaseCode/app/src/main/java/micromaster/galileo/edu/databaseexample/MainActivity.java
package micromaster.galileo.edu.databaseexample;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;

import java.util.ArrayList;

import micromaster.galileo.edu.databaseexample.adapter.ContactAdapter;
import micromaster.galileo.edu.databaseexample.database.DataBaseDAO;
import micromaster.galileo.edu.databaseexample.model.Contact;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private ContactAdapter contactAdapter;
    private ArrayList<Contact> contactArrayList;

    private DataBaseDAO mDataBaseDAO;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mDataBaseDAO = new DataBaseDAO(this);
        mDataBaseDAO.open();

        contactArrayList = getContactsFromDB();

        recyclerView = (RecyclerView) findViewById(R.id.contact_recycler_view);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(layoutManager);

        contactAdapter = new ContactAdapter(contactArrayList);
        recyclerView.setAdapter(contactAdapter);

        final Button button = (Button) findViewById(R.id.add_new_contact);
        button.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent downloadIntent = new Intent(getApplicationContext(), AddNewContactActivity.class);
                startActivity(downloadIntent);
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        contactArrayList.clear();
        contactArrayList.addAll(getContactsFromDB());
        contactAdapter.notifyDataSetChanged();
    }

    private ArrayList<Contact> getContactsFromDB() {
        return mDataBaseDAO.getAllContacts();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mDataBaseDAO.close();
    }
}

File: Android/WeatherApp/app/src/main/java/com/example/robertduriancik/weatherapp/ui/activities/MainActivity.kt
package com.example.robertduriancik.weatherapp.ui.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.example.robertduriancik.weatherapp.R
import com.example.robertduriancik.weatherapp.domain.commands.RequestForecastCommand
import com.example.robertduriancik.weatherapp.ui.adapters.ForecastListAdapter
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.uiThread

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        forecastList.layoutManager = LinearLayoutManager(this)

        doAsync {
            val result = RequestForecastCommand(94043).execute();
            uiThread {
                forecastList.adapter = ForecastListAdapter(result) {
                    startActivity<DetailActivity>(DetailActivity.ID to it.id,
                            DetailActivity.CITY_NAME to result.city)
                }
            }
        }
    }
}
